packages/react/src/primitives/SwitchField/useSwitch.ts

```diff
@@ -0,0 +1,32 @@
+import React, { ChangeEvent, useCallback, useState } from 'react';
+
+export const useSwitch = (props) => {
+  const { onChange, isChecked, defaultChecked, isDisabled } = props;
+  const isControlled = typeof isChecked !== 'undefined';
+  const [isOn, setIsOn] = useState(isControlled ? isChecked : defaultChecked);
```

Comment:
We could switch this to isChecked but then we would have to rearrange the prop that is currently being used for isChecked (this is the prop that turns this into a controlled input and defines the state of the checkbox).  Because  the current usage of isChecked is being exposed to the user and the current implementation of isOn is hidden from the user I thought it would be better in this order. But I am open to switching them.

Reply:
Makes sense